Cabal/src/Distribution/Simple/Build.hs

```diff
@@ -145,10 +147,21 @@ build pkg_descr lbi flags suffixes = do
             , withPackageDB = withPackageDB lbi ++ [internalPackageDB]
             , installedPkgs = index
             }
+    par_strat <-
+      toFlag <$> case buildUseSemaphore flags of
+        Flag sem_name -> case buildNumJobs flags of
+          Flag{} -> do
+            warn verbosity $ "Ignoring -j due to --semaphore"
+            return $ UseSem sem_name
```

Comment:
`UseSem sem_name` in both branches, maybe it can be put out of the `case`?

Reply:
We could probably with `flagToMaybe` and `when` but it would probably compile down to cases in the end :shrug: